### ISSUE TITLE
Support for custom objective in lightgbm and xgboost

### ISSUE THREAD

**Issue by pcgm-team** [functime-org/functime#178]
This is already supported in FLAML. See here: https://microsoft.github.io/FLAML/docs/Examples/AutoML-for-LightGBM

**Comment by baggiponte** [comment]
Ciao @pcgm-team, sorry for being so late, really been busy with work. Will look at this later. I did not implement the FLAML code, so I should invoke @topher-lo. Can you make an example of a custom objective, is it a custom loss to train the model on?
